pkg/storage/fs/ocis/upload.go

```diff
@@ -129,6 +138,18 @@ func (fs *ocisfs) InitiateUpload(ctx context.Context, ref *provider.Reference, u
 		if _, ok := metadata["sizedeferred"]; ok {
 			info.SizeIsDeferred = true
 		}
+		if metadata["checksum"] != "" {
+			parts := strings.SplitN(metadata["checksum"], " ", 2)
+			if len(parts) != 2 {
+				return nil, errtypes.BadRequest("invalid checksum format. must be '[algorithm] [checksum]'")
```

Comment:
this error is missleading, according to your comment:

```xml
<oc:checksums>
  <oc:checksum>SHA1:941f6bb205d2e9f MD5:aeoaeoaeoaeoaeo ADLER32:iaeoioeao</oc:checksum>
</oc:checksums>
```

there could be the case that a client sends more than one tuple of `<algo:hash>` that is not being accounted for. We should either:

- check that and throw an error because we don't support it yet
- add a check to deal with this case

Reply:
I am following the TUS `Upload-Checksum` header internally ... and that is single value ... anyway. Part of https://github.com/cs3org/cs3apis/issues/103 I'd say because currently this is all in Opaque headers.